Migrated messages into a context.

G3sit opened the pull request:
- Created a MessageContext => no need to call reload_app anymore.
- Included MessageContext into the normal App Hierarchy.
- Replaced all old usages with the new usage
  - To keep changes minimal, withMessageContext and MessageContextConsumer is used. It is planed to use Hooks instead, while migrating to function components.

TODO

- [x] Find a solution for usage in Exporter

G3sit committed:
ported messages to a context and function component.

Currently untested

G3sit committed:
added very basic test of MyProfile page

G3sit committed:
removed unused import

G3sit committed:
added a back button test to myprofile page.

G3sit committed:
added test of currently misbehaving feature

G3sit committed:
removed code for password confirmation of my profile page.

Fixed missing not same password warning

G3sit committed:
moved last opened saves to own component and moved tests to it.

G3sit committed:
fixed warnings

G3sit committed:
converted my profile page into a function component

G3sit committed:
made export promise reject on validation error

nic-schi commented on src/components/platform/logout/Logout.tsx:
```diff
@@ -3,23 +3,24 @@ import {Session} from "../../../general-components/Session/Session";
             reload_app();
 
-            Messages.add("Bis bald!", "SUCCESS", Messages.TIMER);
+            addMessage("Bis bald!", "SUCCESS", Messages.TIMER);
```
Message is shown twice while user logs out. also this seems to be a bigger problem (maybe inside loader?)

G3sit replied:
The issues appears, because the developer build of React is used. Since React 18, the Strict mode does mount und unmount every component once before mounting it for real. More information about that [here](https://react.dev/reference/react/StrictMode#fixing-bugs-found-by-re-running-effects-in-development). 

I don't think it is useful to fix that here. To be precise, in my opinion, the only useful fix would be to remove the feature of the Loader to call the payload functions. I would change the payload property to accept a promise object, which it subscribes to.

G3sit replied:
Okay, I didn't read the code of the logoutCallback carefully enough. Since the function does perform a network request, I will try to add the above feature without removing the normal payload feature.

G3sit replied:
I added the described feature, but realized that sending the Logout request in componentDidMount or with a function component useEffect, isn't so useful, because it has nothing to do with the visual representation. As a result, I migrated the Logout page to only serve as loader and redirect page.

nic-schi commented on src/general-components/Utility/Hooks.ts:
```diff
@@ -0,0 +1,26 @@
+import {useCallback, useState} from "react";
+
+/**
+ * Hook und ein boolean state zu erstellen. Erstellt höufig genutzen callbacks.
```
Rechtschreibfehler anpassen.

1. ... "um ein"... 
2. höufig

ma1160 reviewed (changes requested):
siehe nichlas

nic-schi reviewed (changes requested):
Double requests and grammatic errors

G3sit committed:
fixed typos.

G3sit committed:
migrate Loader to function component and fixed logout behaviour.

The Logout code was moved to Nav because it is unrelated to Rendering in any way. Additionally logout requests should only be send once, which would not be possible when callend in onEffect.

G3sit committed:
wrapped redirect in useEffect.
